bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -857,7 +884,13 @@ public synchronized void start() {
 
         ledgerStorage.start();
 
-        syncThread.start();
+        // check the bookie status to start with
+        if (forceReadOnly.get()) {
```

Comment:
yes, forceReadOnly will override status from disk. The forceReadOnly is set when setForceReadOnly is configured. That is documented in service configuration.